### PR TITLE
positionIndex can now return to 0

### LINKED ISSUE #26: positionIndex cannot return to 0 after it changes
Given this, it is not possible to return the list to the top since it is zero-indexed.
```hbs
{{input value=customIndex}}
{{#virtual-each ... positionIndex=customIndex ... }}
```

It looks like `startingIndex` in `scrollTo` just needs to allow for this.  What do you think of this?
```js
  scrollTo(positionIndex) {
    const itemHeight = this.getAttr('itemHeight');
    const totalHeight = get(this, '_totalHeight');
    const _visibleItemCount = get(this, '_visibleItemCount');

   // const startingIndex = Math.max(positionIndex, 0) || get(this, '_startAt');
    var startingIndex = Math.max(positionIndex, 0);
    if (isNaN(startingIndex)) {
      startingIndex = get(this, '_startAt');
    }

    const maxVisibleItemTop = Math.max(0, (get(this, '_items.length') - _visibleItemCount + EXTRA_ROW_PADDING));
    const maxPadding = Math.max(0, totalHeight - ((_visibleItemCount - 1) * itemHeight) + (EXTRA_ROW_PADDING * itemHeight));
    const sanitizedIndex = Math.min( startingIndex, maxVisibleItemTop);
    const sanitizedPadding = (sanitizedIndex === maxVisibleItemTop) ? maxPadding : itemHeight * sanitizedIndex;

    this.calculateVisibleItems(sanitizedIndex);
    this.$().scrollTop(sanitizedPadding);
  },
```

### PR DESCRIPTION
**Pull request by bitsoflogic**
Fixes #26

### CODE REVIEW

File: addon/components/virtual-each.js [hunk 1]
```diff
@@ -145,7 +144,7 @@ const VirtualEachComponent = Component.extend(EventListenerMixin, DefaultAttrsMi
     emberRun(() => {
       const startAt = get(this, '_startAt');
       const scrolledAmount = this.$().scrollTop();
-      const visibleStart = positionIndex || Math.floor(scrolledAmount / this.getAttr('itemHeight'));
+      const visibleStart = isNaN(positionIndex) ? Math.floor(scrolledAmount / this.getAttr('itemHeight')) : positionIndex;
```
Comment (flexyford) [inline addon/components/virtual-each.js]: Can we change this line to how we set the starting index as well, ie.

```const visibleStart = isNaN(positionIndex) ? Math.floor(scrolledAmount / this.getAttr('itemHeight')) : Math.max(positionIndex, 0);```

**Comment by bitsoflogic** [comment]
I have the fix pending for #27 as well. I can push it over whenever.  Was holding off until this is approved.

**Comment by flexyford** [comment]
Please break #27 into a separate PR

**Comment by jasonmit** [comment]
Thanks @bitsoflogic and @flexyford 

@flexyford version when you're ready

**Comment by flexyford** [comment]
@jasonmit I'm working on a `virtual-each` issue before versioning. Documentation to come later today hopefully

**Comment by jasonmit** [comment]
:+1:

**Pull request merged**